### PR TITLE
Make `file.path` always return the absolute path

### PR DESCRIPTION
**Pull request by jeremiah-corrado**
Implements the change to `file.path` discussed here: https://github.com/chapel-lang/chapel/issues/20271. 

The method no longer returns the relative path when available; only the absolute path is returned. This PR also adds several tests to validate the method's behavior and improves its documentation.

- [x] paratest passing

### CODE REVIEW

**Review by benharsh — COMMENTED** [review]
The code changes look good to me, but we should have a deprecation warning since the underlying behavior is changing. If you add a config-param, you can use it with where-clauses to have one version of the method with a deprecation warning. Something like:

```chpl
config param filePathAbsolute = false;

// original
deprecated "blah blah blah"
proc file.path : string throws where !filePathAbsolute { ... }

// new version
proc file.path : string throws where filePathAbsolute { ... }
```

File: test/library/standard/IO/filePath/basic.prediff [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+
+# swaps occurences of $CHPL_HOME in the output with the string: "{CHPL_HOME}"
+
+tmpfile=$2
+
+tmptmp=`mktemp "tmp.XXXXXX"`
+
+regex="s~$CHPL_HOME~{CHPL_HOME}~g"
+sed -e "$regex" $tmpfile > $tmptmp
```
[the remark this answers is not in the document]
Comment (jeremiah-corrado) [reply test/library/standard/IO/filePath/basic.prediff:10]: Yeah, this script works on my laptop and chapcs.

**Comment by jeremiah-corrado** [comment]
Oh true, not sure how I missed that, I'll add a param to switch between implementations. Thanks for pointing that out!

**Comment by jeremiah-corrado** [comment]
Ok, I added a `config param` to control the relative/absolute path behavior. There is a deprecation test to go along with it.

This also involved including a private proc that supports the old behavior which had the nice side effect of printing relative file paths in IO error messages. (So my initial changes to `types/file/freadIntFailed` and `types/file/fwriteIntFailed` were reverted).

File: modules/standard/IO.chpl [hunk 2]
```diff
@@ -1631,50 +1631,86 @@ proc file.fsync() throws {
   if err then try ioerror(err, "in file.fsync", this._tryGetPath());
 }
 
+/*
+  A compile-time parameter to control the behavior of :proc:`file.path`
+
+  When 'false', the deprecated behavior is used (i.e., return the relative path
+  whenever available)
```
Comment (benharsh) [inline modules/standard/IO.chpl:1638]: Technically I believe it's the "shortest" path between the relative and absolute paths.

**Review by benharsh — APPROVED** [review]
Looks good to me!

File: modules/standard/IO.chpl (hunk 2 above)
Comment (jeremiah-corrado) [reply modules/standard/IO.chpl:1638]: Good point, I'll fix that wording. Thanks!

**Pull request merged**